qiime2/metadata.py

```diff
@@ -14,6 +14,38 @@
 
 class Metadata:
     def __init__(self, dataframe):
+        # `/` and `\0` aren't permitted because they are invalid filename
+        # characters on *nix filesystems. The remaining values aren't permitted
+        # because they *could* be misinterpreted by a shell (e.g. `*`, `|`).
+        illegal_chars = ['/', '\0', '\\', '*', '<', '>', '?', '|', '$']
+        illegal_chars = set(illegal_chars)
+        chars_for_msg = ", ".join("%r" % i for i in illegal_chars)
+
+        categories = dataframe.columns.values.tolist()
+        indices = dataframe.index.tolist()
+
+        if dataframe.columns.dtype_str != 'object':
```

Comment:
til about `dtype_str`, but what does it do? I can't find any docs on it. I assume it lists `str` types as `object`?